src/api/getRecipe.js

```diff
@@ -1,6 +1,13 @@
 const getRecipe = (id) =>
-  fetch(`http://localhost:8080/recipe/${id}`).then((response) =>
-    response.json()
-  );
+  fetch(`http://localhost:8080/recipe/${id}`)
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
+      }
+      return response.json();
+    })
+    .catch((err) => {
+      throw new Error(err);
+    });
```

Comment:
Unlike `axios` or else, the `fetch` callback doesn't throw an Error - adding that so the React ("Tanstack") Query can handle error properly (by default, it will attempt retries 3 times).